tests/visualization_tests/test_contour.py

```diff
@@ -53,65 +194,80 @@ def test_target_is_not_none_and_study_is_multi_obj() -> None:
         None,
     ],
 )
-def test_plot_contour(params: Optional[List[str]]) -> None:
+def test_get_contour_info_empty(
+    specific_create_study: Callable[[], Study], params: Optional[List[str]]
+) -> None:
 
-    # Test with no trial.
-    study_without_trials = prepare_study_with_trials(no_trials=True)
-    figure = plot_contour(study_without_trials, params=params)
-    assert len(figure.data) == 0
+    study = specific_create_study()
+    info = _get_contour_info(study, params=params)
+    assert len(info.sorted_params) == 0
+    assert len(info.sub_plot_infos) == 0
 
-    # Test whether trials with `ValueError`s are ignored.
-    study = create_study()
-    study.optimize(fail_objective, n_trials=1, catch=(ValueError,))
-    figure = plot_contour(study, params=params)
-    assert len(figure.data) == 0
 
-    # Test with some trials.
+def test_get_contour_info_error() -> None:
     study = prepare_study_with_trials()
 
-    # Test ValueError due to wrong params.
     with pytest.raises(ValueError):
-        plot_contour(study, ["optuna", "Optuna"])
+        _get_contour_info(study, ["optuna", "Optuna"])
 
-    figure = plot_contour(study, params=params)
-    if params is not None and len(params) < 3:
-        if len(params) <= 1:
-            assert not figure.data
-        elif len(params) == 2:
-            assert figure.data[0]["x"] == (0.925, 1.0, 2.5, 2.575)
-            assert figure.data[0]["y"] == (-0.1, 0.0, 1.0, 2.0, 2.1)
-            assert figure.data[0]["z"][3][1] == 0.0
-            assert figure.data[0]["z"][2][2] == 1.0
-            assert figure.data[0]["colorbar"]["title"]["text"] == "Objective Value"
-            assert figure.data[1]["x"] == (1.0, 2.5)
-            assert figure.data[1]["y"] == (2.0, 1.0)
-            assert figure.layout["xaxis"]["range"] == (0.925, 2.575)
-            assert figure.layout["yaxis"]["range"] == (-0.1, 2.1)
-    else:
-        # TODO(crcrpar): Add more checks. Currently this only checks the number of data.
-        n_params = len(params) if params is not None else 4
-        assert len(figure.data) == n_params**2 + n_params * (n_params - 1)
+
+@pytest.mark.parametrize("params", [[], ["param_a"]])
+def test_get_contour_info_too_short_params(params: Optional[List[str]]) -> None:
```

Comment:
No `None` in this parameterisation:
```suggestion
def test_get_contour_info_too_short_params(params: List[str]) -> None:
```

Otherwise,  how about adding `None` as a value of `params`?